File: 박인우/[프로그래머스] 오픈 채팅방.py
def solution(record):
    answer = []
    nickname = {}
    for info in record:
        info = info.split()
        if info[0] == "Enter":
            nickname[info[1]] = info[2]
            answer.append((info[1], "님이 들어왔습니다."))
        if info[0] == "Leave":
            answer.append((info[1], "님이 나갔습니다."))
        if info[0] == "Change":
            nickname[info[1]] = info[2]
    return [nickname[i[0]]+i[1] for i in answer]

File: 박인우/0207-0213/[프로그래머스] 오픈채팅방.swift
import Foundation

func solution(_ record:[String]) -> [String] {
    
    var idDict: [String: String] = [:]
    var log: [[String]] = []
    
    for s in record {
        let lst = s.components(separatedBy: " ")
        
        if lst[0] == "Enter" || lst[0] == "Change" {
            idDict[lst[1]] = lst[2]
        }
        
        if !(lst[0] == "Change") {
            log.append([lst[0], lst[1]])
        } 
    }
    
    return log.map {
        if $0[0] == "Enter" {
            return "\(idDict[$0[1]]!)님이 들어왔습니다."
        } else {
            return "\(idDict[$0[1]]!)님이 나갔습니다."
        }
    }
}
